samtranslator/model/eventsources/cloudwatchlogs.py

```diff
@@ -47,5 +47,7 @@ def get_subscription_filter(self, function, permission):
         subscription_filter.LogGroupName = self.LogGroupName
         subscription_filter.FilterPattern = self.FilterPattern
         subscription_filter.DestinationArn = function.get_runtime_attr("arn")
+        if 'Condition' in function.resource_attributes:
```

Comment:
Is there a way we can not have to check this in every event source (and all new event sources moving forward)? Is there something we can do that's more DRY? I understand for certain event sources, we may need to override and do something different, e.g., S3. If there's no obvious way to do this, then I'm ok leaving it this way.

Reply:
I think so, I'll look into it when I get a chance.